### PR TITLE
EVG-13401 edit annotations permissions

### PR DESCRIPTION
**Pull request by ablack12**
This allows annotation access to be given at the project level, and it will integrate with mana correctly, as the new permission will be part of the task dropdown.
image

Are we okay with the ID of the role being randomly generated through here? https://github.com/evergreen-ci/gimlet/blob/master/rolemanager/role_manager.go#L912
Because Mana requires Project Settings / Patches / Logs to also be defined, there's the risk of multiple different kind of roles being requested along with the task annotations level (although probably only project settings is the one that would differ), so completing a migration to add named roles seems unhelpful.

### CODE REVIEW

File: rest/route/service.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	editRoles := RequiresSuperUserPermission(evergreen.PermissionRoleModify, evergreen.RoleModify)
 	viewTasks := RequiresProjectPermission(evergreen.PermissionTasks, evergreen.TasksView)
 	editTasks := RequiresProjectPermission(evergreen.PermissionTasks, evergreen.TasksBasic)
+	// editAnnotations := RequiresProjectPermission(evergreen.PermissionTasks, evergreen.TasksAnnotation)
```
Comment (ablack12) [inline rest/route/service.go:42]: commenting out because the PUT routes aren't finished yet so this won't be used yet

**Review by john-m-liu — COMMENTED** [review]
I know we talked about this before, but what if we make editing annotations separate from the rest of task permissions? I'm just paranoid that in the future we may want to have some people (build barons?) who can edit annotations but not be able to set high priorities or override dependencies

**Comment by ablack12** [comment]
We would still need Mana's help to do that, but it may make sense, I can reach out to them. In the case that they won't be able to make the change on their end for some time, I would probably argue to stick with this.

**Comment by ablack12** [comment]
Okay mana implies this new dropdown should work for them. To make sure that existing users have task annotations permissions, we'll want to add `"project_task_annotations":10` to the basic_project_access role (or we could do this for anything matching the query `{"permissions.project_tasks": {"$gt":0}}` to also hit project admins, but everybody has the basic project access role so it's maybe not necessary. We'd also add `"project_task_annotations": 20` to admin_project_access.

File: rest/route/service.go [hunk 2]
```diff
@@ -192,5 +194,5 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/versions/{version_id}/abort").Version(2).Post().Wrap(checkUser, editTasks).RouteHandler(makeAbortVersion(sc))
 	app.AddRoute("/versions/{version_id}/builds").Version(2).Get().Wrap(viewTasks).RouteHandler(makeGetVersionBuilds(sc))
 	app.AddRoute("/versions/{version_id}/restart").Version(2).Post().Wrap(checkUser, editTasks).RouteHandler(makeRestartVersion(sc))
-	app.AddRoute("/versions/{version_id}/annotations").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeFetchAnnotationsByVersion(sc))
+	app.AddRoute("/versions/{version_id}/annotations").Version(2).Get().Wrap(checkUser, viewAnnotations).RouteHandler(makeFetchAnnotationsByVersion(sc))
```
Comment (malikchaya2) [inline rest/route/service.go:197]: I have this https://github.com/evergreen-ci/evergreen/pull/4188 in code review that adds a route. We should figure out which one gets merged first, and modify the other.

**Review by malikchaya2 — APPROVED** [review]

File: rest/route/service.go (hunk 2 above)
Comment (ablack12) [reply rest/route/service.go:197]: yep, will DM

File: globals.go [hunk 3]
```diff
@@ -666,6 +667,18 @@ var (
...
 	}
+	AnnotationsModify = PermissionLevel{
+		Description: "Modify annotations",
+		Value:       20,
+	}
+	AnnotationsView = PermissionLevel{
+		Description: "View annotations",
+		Value:       10,
+	}
+	AnnotationsNone = PermissionLevel{
```
Comment (john-m-liu) [inline globals.go:678]: is there a scenario where someone shouldn't even be able to view annotations?

Comment (ablack12) [reply globals.go:678]: Not currently. But we do have this same option for logs, patches, distros, hosts, tasks, and project settings (i.e. I think every other permission) so it seemed consistent to define it.

Comment (ablack12) [reply globals.go:678]: Non users obviously shouldn't be able to view, so even if we didn't explicitly define this permission it is implicit.

Comment (john-m-liu) [reply globals.go:678]: oh good point, not logged in users can see the task page but probably shouldn't be able to see annotations

**Review by john-m-liu — APPROVED** [review]

**Pull request merged**